### PR TITLE
Feature/redux api

### PR DESCRIPTION
**Pull request by Obote**
### **What does this PR do?**

- Utilized the Bookstore API as the backend. 
- Employed axios with createAsyncThunk to handle HTTP requests to the backend. 
- Fetched books from the Bookstore API after the component had rendered. 
- Refactored the add book and remove book features to persist changes in the server. 
- Ensured that the application behaved the same way after the refactoring.

### CODE REVIEW

File: src/redux/books/bookSlice.js [hunk 1]
```diff
@@ -1,42 +1,109 @@
-import { createSlice } from '@reduxjs/toolkit';
+/* eslint-disable import/no-extraneous-dependencies */
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
```
Comment (anagudelogu) [inline src/redux/books/bookSlice.js]: - [ ] Please avoid disabling linters as they are very useful to keep a consistent code format and ensure the use of best practices. Instead, please consider solving the errors! 🤓 🔧

File: src/redux/books/bookSlice.js [hunk 2]
```diff
@@ -1,42 +1,109 @@
...
+        title,
+        author,
+        category: 'Non fiction',
+      });
+      return dataStream;
+    } catch (error) {
+      return error;
+    }
+  },
+);
```
Comment (anagudelogu) [inline src/redux/books/bookSlice.js]: - [ ] I'm getting the following error trying to create a new book:

```json
{"error":{"status":400,"message":"'item_id' is required."}}
```

it seems to be a typo in `itemId` it should be `item_id`. Please consider applying this change! 🤓 🔧

**Review by anagudelogu — CHANGES_REQUESTED** [review]
# STATUS:  CHANGES REQUIRED  ♻️ 🚧 

Hi @Obote, 
I'm Andres.


While you have done a great job working on this milestone 💻 , there are some issues that you still need to work on to go to the next project. ♻️ 

You are almost there, keep up the good work! 🚀 



## Highlights
✅ Great use of Git flow. 💯 
✅ Your project is documented professionally. 📜 
✅ Descriptive PR title, summary and commit messages. 💬 

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by KayLemba — APPROVED** [review]
# STATUS: APPROVED 🟢 
Hello @Obote    ,
Your project is complete! There is nothing else to say other than... it's time to merge it  .
Congratulations! 🎉 Keep up the Awesome work 🥂 Keep Rocking 🥇

HIGHLIGHTS:

- [x]  Your repo is professional 😎 

- [x] Descriptive P.R 💯 Professional README file

- [x] App runs as intended



Now you can merge this branch and move on 🥇
Happy Coding cheers 🥇 💯

Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using

**Pull request merged**